service_capacity_modeling/interface.py

```diff
@@ -634,6 +640,9 @@ class CapacityDesires(ExcludeUnsetModel):
     # What will the state look like
     data_shape: DataShape = DataShape()
 
+    # What is the current microarchitectural/system configuration of the system
+    current_capacity: CurrentCapacity = CurrentCapacity()
```

Comment:
This should probably be Optional[CurrentCapacity] = None. That way you don't have to have default values.